src/entities/addLiquidity/weighted/addLiquidityWeightedV3.ts

```diff
@@ -0,0 +1,168 @@
+import { Token } from '../../token';
+import { TokenAmount } from '../../tokenAmount';
+import { BALANCER_VAULT, MAX_UINT256, ZERO_ADDRESS } from '../../../utils';
+import {
+    AddLiquidityBase,
+    AddLiquidityBuildOutput,
+    AddLiquidityInput,
+    AddLiquidityKind,
+    AddLiquidityBaseQueryOutputV3,
+    AddLiquidityWeightedV3Call,
+} from '../types';
+import { AddLiquidityAmounts, PoolState } from '../../types';
+import {
+    doAddLiquidityQueryV3,
+    getAmounts,
+    parseAddLiquidityArgsV3,
+} from '../../utils';
+
+export class AddLiquidityWeightedV3 implements AddLiquidityBase {
```

Comment:
With the invariant approach in v3, we won't need this distinction anymore. We can have a generalized `AddLiquidityV3` that can cover all pool types (minus custom scenarios).